src/Storages/MergeTree/MergeTreePartition.cpp

```diff
@@ -152,29 +161,78 @@ void MergeTreePartition::store(const Block & partition_key_sample, const DiskPtr
     auto out = disk->writeFile(part_path + "partition.dat");
     HashingWriteBuffer out_hashing(*out);
     for (size_t i = 0; i < value.size(); ++i)
-        partition_key_sample.getByPosition(i).type->getDefaultSerialization()->serializeBinary(value[i], out_hashing);
+    {
+        if (partition_key_sample.getByPosition(i).name.starts_with("modulo"))
```

Comment:
But why we change the serialization of the new `modulo`? I thought that we should be able to read the old one, but the new `modulo` should work as expected.